README.md

```diff
@@ -1,28 +1,126 @@
 # Extreme Classification [![Build Status](https://travis-ci.org/vishwakftw/extreme-classification.svg?branch=master)](https://travis-ci.org/vishwakftw/extreme-classification)
 
-Project Repository for CS6370 : Information Retrieval offered in Fall 2018
+`extreme_classification` is a Python module designed for extreme classification tasks with two new algorithms:
+- **NeuralXC**: A deep-learning based solution using autoencoders and neural networks.
 
-`extreme_classification` is a Python module designed for the project.
+- **HierarchicalXC**: A hierarchical clustering based approach
 
-`train_neuralxc.py` is a Python script used to train an extreme classification model using **NeuralXC** which is an autoencoder-regression based model for this task.
+## Using the module
 
-## Setup
+### Setup
 
-Requirements:
+- Requirements for the project are listed in `requirements.txt`. In addition to these, it is mandatory to have PyTorch installed in your system with version 0.4.1 or higher.
+   - Requirements can be installed using `pip`:
+     ```bash
+     $ pip install -r requirements.txt 
+     ```
+     or using `conda`:
+     ```bash
+     $ conda install --file requirements.txt
+     ```
 
-- Python 2.7 or 3.5
-- PyTorch 0.4.1 or above
-- NumPy
-- SciPy
-- Matplotlib
-- Scikit-Learn
-- PyYAML
+### Installing `extreme_classification`
 
-Requirements can be installed using `pip`:
+- Clone the repository.
+
+- Run `python setup.py install`.
+
+- To test if your installation is successful, try running the command:
+```bash
+$ python -c "import extreme_classification"
+```
+
+## Using the training scripts
+
+- `train_neuralxc.py` is a Python script used to train an extreme classification model using **NeuralXC**. You can run the script by passing required options:
 ```bash
-$ pip install -r requirements.txt 
+python train_neuralxc.py [-h] --data_root DATA_ROOT --dataset_info DATASET_INFO

```

Comment:
Why do we need this? Isn't message to use `--help` enough? If we modify options, we will have to change README each time

Reply:
Fair enough